### ISSUE TITLE
Specialised categories of tokens within a token type

### ISSUE THREAD

**Issue by Sidnioulz** [Sidnioulz/design-tokens-docs#6]
Some token systems define subcategories of tokens beyond types recognised by current dominant tools or the DTCG spec.

## Examples

In the examples below, in parentheses are names of design token documentations that use a tailor-made preview for this category.

### Color
* Background color
* Foreground/text color (Back Market)
* Border color (Back Market)
* Gradients (Datadog Druids for disabled scrim)

Size can be mapped to
* Width and height (Polaris)
* Margin (France Télévisions)
* Gap (France Télévisions)
* Border width (France Télévisions)
* Generic size

To what extent should a general documentation tool support previews for such categories within a type? Should a bunch of presets be proposed? If so, how would a user of the doc tool map specific tokens to a category?

**Comment by drwpow** [comment]
I’ve seen a lot of different framings of what is codified in tokens vs what aren’t, and what these are named. If you look at the spread of most token systems, most design tokens are color, followed by typography. So it’s more important to categorize and navigate the bulk of the DS.

Further, the concept of “foreground” and “background” is a pretty standard concept that exists in every DS, no matter what it’s named. So I think there are some characteristics that would apply universally. But the hierarchy is difficult, because some systems would have foreground/background as the top “tier,” whereas other systems may rely on “elevation” or “layers” or “primary/secondary/tertiary” contexts that foreground/background rest within.

**Suggestion**

Rather than a formal hierarchy, a tagging system is pretty robust. For example, someone could tag a token with `foreground` and `elevation.1` without the docs having any idea of how those concepts relate to one another. I think this could solve all the problems mentioned. As for “preset” values, I’m not sure? Perhaps some starter set of “tags” would be helpful just so people get the general idea! But as long as they’re just starter examples without being “baked in” that would solve most needs, I think.

**Comment by Sidnioulz** [comment]
You're right that elements of the hierarchy should ideally not influence how an individual token is previewed, as the order of hierarchy differs between systems.

This being said, the partial audits done show that several tailor-made docs sites alter their doc layouts based on the type of token being rendered. Grouping together tokens with the same optimal preview can lead to better doc layouts.

So, there could be value in switching to more specialised previews / token doc table layouts when possible without muddying the hierarchy chosen by the token maintainers.

----

There would also be value in using a tag-based approach to complement the types that end up being codified in DTCG formats. It would also support token pickers or quick filters / advanced search features to have such tagging.

I feel that if some tags enable a different preview UI or overall token row layout, though, then users would quickly request the ability to render their own token rows for their own tokens.

I'm going to revisit this issue once most of the UI preview tickets have reached a consensus stage, once we get a better sense of how much really is gained by seeking to group tokens by "type" to provide more consistent layouts.

Curious to hear more of your thoughts on the topic.
